torchgeo/datasets/agrifieldnet.py

```diff
@@ -0,0 +1,276 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""AgriFieldNet India Challenge dataset."""
+
+import os
+import re
+from collections.abc import Iterable, Sequence
+from typing import Any, Callable, Optional, Union, cast
+
+import matplotlib.pyplot as plt
+import torch
+from matplotlib.figure import Figure
+from rasterio.crs import CRS
+from torch import Tensor
+
+from .geo import RasterDataset
+from .utils import BoundingBox, RGBBandsMissingError
+
+
+class AgriFieldNet(RasterDataset):
+    """AgriFieldNet India Challenge dataset.
+
+    The `AgriFieldNet India Challenge
+    <https://zindi.africa/competitions/agrifieldnet-india-challenge>`__ dataset
+    includes satellite imagery from Sentinel-2 cloud free composites
+    (single snapshot) and labels for crop type that were collected by ground survey.
+    The Sentinel-2 data are then matched with corresponding labels.
+    The dataset contains 7081 fields, which have been split into training and
+    test sets (5551 fields in the train and 1530 fields in the test).
+    Satellite imagery and labels are tiled into 256x256 chips adding up to 1217 tiles.
+    The fields are distributed across all chips, some chips may only have train or
+    test fields and some may have both. Since the labels are derived from data
+    collected on the ground, not all the pixels are labeled in each chip.
+    If the field ID for a pixel is set to 0 it means that pixel is not included in
+    either of the train or test set (and correspondingly the crop label
+    will be 0 as well). For this challenge train and test sets have slightly
+    different crop type distributions. The train set follows the distribution
+    of ground reference data which is a skewed distribution with a few dominant
+    crops being over represented. The test set was drawn randomly from an area
+    weighted field list that ensured that fields with less common crop types
+    were better represented in the test set. The original dataset can be
+    downloaded from `Source Cooperative <https://beta.source.coop/
+    radiantearth/agrifieldnet-competition/>`__.
+
+    Dataset format:
+
+    * images are 12-bands Sentinel-2 data
+    * masks are tiff image with unique values representing the class and field id
+
+    Dataset classes:
+
+    0 - No-Data
+    1 - Wheat
+    2 - Mustard
+    3 - Lentil
+    4 - No Crop/Fallow
+    5 - Green pea
+    6 - Sugarcane
+    8 - Garlic
+    9 - Maize
+    13 - Gram
+    14 - Coriander
+    15 - Potato
+    16 - Bersem
```

Comment:
Should this be Berseem?

Reply:
Berseem is more reasonable. The [competition website](https://zindi.africa/competitions/agrifieldnet-india-challenge/data) and [GitHub](https://github.com/radiantearth/agrifieldnet_india_competition) both said Bersem. But I can't find much information about Bersem. I prefer Berseem for now.